src/main/java/com/faforever/client/preferences/LanguageInfo.java

```diff
@@ -0,0 +1,80 @@
+package com.faforever.client.preferences;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+public enum LanguageInfo {
+  AUTO("auto", null, "System", 0), EN("en", "UK", "English", 1), DE("de", "DE", "Deutsch", 2), RU("ru", "RU", "ру́сский", 3);
+
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final String languageCode;
+  private final String countryCode;
+  private final String dispayName;
+  private final int position;
+
+  LanguageInfo(String languageCode, String countryCode, String displayName, int position) {
+    this.countryCode = countryCode;
+    this.dispayName = displayName;
+    this.languageCode = languageCode;
+    this.position = position;
+  }
+  public static LanguageInfo getLanuageInfoByLanguageCode(String code) {
+    for (LanguageInfo LI : LanguageInfo.values()) {
```

Comment:
Variable names always start lower case. And:
> Prefer longer, unambiguous names over short but ambiguous ones

In this case `languageInfo` or `info`